tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_sms_notification.py

```diff
@@ -18,12 +18,19 @@ def test_sms_notification(self):
 
         system = System(self.marionette)
 
+        self.assertEqual(0, len(self.data_layer.get_all_sms()))
+
         # Send a SMS to the device
         self.data_layer.send_sms(self.environment.phone_numbers[0], _text_message_content)
 
-        # We will wait upto 300 seconds for the SMS to arrive due to network latency
-        system.wait_for_notification_toaster_displayed(timeout=300,
-                    message="Notification did not appear. SMS database dump: %s " % self.data_layer.get_all_sms())
-        system.wait_for_notification_toaster_not_displayed()
+        # self.data_layer.send_sms can be so slow that the sms has already been received,
+        # That's why the following if statement, see also bug 1172460
+        if len(self.data_layer.get_all_sms()) == 1:
```

Comment:
I think it's better to wait for the device to have at least  1 SMS and then check the notification. In the current test, we might not test the presence of the notification which makes the test name lying.

``` python
Wait(self.marionette).until(lamba m: self.data_layer.get_all_sms() >= 1)
system.wait_for_notification_toaster_displayed(timeout=300,
    message="Notification did not appear. SMS database dump: %s " % self.data_layer.get_all_sms())
system.wait_for_notification_toaster_not_displayed()
self.assertEqual(2, len(self.data_layer.get_all_sms()))
```